src/main/java/de/tum/in/www1/artemis/repository/StudentParticipationRepository.java

```diff
@@ -643,30 +644,59 @@ List<StudentParticipation> findByExerciseIdWithLatestSubmissionWithoutManualResu
             """)
     List<StudentParticipation> findAllWithEagerSubmissionsAndEagerResultsAndEagerAssessorByExerciseIdIgnoreTestRuns(@Param("exerciseId") long exerciseId);
 
-    @Query(value = """
-            SELECT p
+    @Query("""
+            SELECT p.id
             FROM StudentParticipation p
-                LEFT JOIN FETCH p.submissions s
-                LEFT JOIN FETCH p.results r
             WHERE p.exercise.id = :exerciseId
-                AND (
-                    p.student.firstName LIKE %:partialStudentName%
-                    OR p.student.lastName LIKE %:partialStudentName%
-                ) AND r.completionDate IS NOT NULL
-            """, countQuery = """
+              AND (
+                  p.student.firstName LIKE %:partialStudentName%
+                  OR p.student.lastName LIKE %:partialStudentName%
+              ) AND EXISTS (
+                  SELECT 1
+                  FROM Result r
+                  WHERE r.participation.id = p.id
+                    AND r.completionDate IS NOT NULL
+              )
```

Comment:
Is this subquery actually necessary? I think left joins should work here as well